File: samples/javascript/apidocs/ee-array-tan.js
/**
 * Copyright 2021 The Google Earth Engine Community Authors
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

// [START earthengine__apidocs__ee_array_tan]
var π = Math.PI;
print(ee.Array([-π / 4]).tan());  // [Almost -1]
print(ee.Array([0]).tan());  // [0]
print(ee.Array([π / 4]).tan());  // [Almost 1]

var start = -π / 3;
var end = π / 3;
var points = ee.Array(ee.List.sequence(start, end, null, 50));
var values = points.tan();

// Plot tan() defined above.
var chart = ui.Chart.array.values(values, 0, points)
    .setOptions({
      viewWindow: {min: start, max: end},
      hAxis: {
        title: 'x',
        viewWindowMode: 'maximized',
        ticks: [
          {v: start, f: '-π / 3'},
          {v: 0, f: 0},
          {v: end, f: 'π / 3'}]
      },
      vAxis: {
        title: 'tan(x)',
        ticks: [
          {v: -Math.sqrt(3), f: '-√3'},
          {v: 0},
          {v: Math.sqrt(3), f: '√3'}]
      },
      lineWidth: 1,
      pointSize: 0,
    });
print(chart);
// [END earthengine__apidocs__ee_array_tan]

File: samples/python/apidocs/ee_array_tan.py
# Copyright 2022 The Google Earth Engine Community Authors
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#    https://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.

# [START earthengine__apidocs__ee_array_tan]
import math
import altair as alt
import pandas as pd

π = math.pi
display(ee.Array([-π / 4]).tan())  # [Almost -1]
display(ee.Array([0]).tan())  # [0]
display(ee.Array([π / 4]).tan())  # [Almost 1]

start = -π / 3
end = π / 3
points = ee.Array(ee.List.sequence(start, end, None, 50))
values = points.tan()

df = pd.DataFrame({'x': points.getInfo(), 'tan(x)': values.getInfo()})

# Plot tan() defined above.
alt.Chart(df).mark_line().encode(
    x=alt.X('x', axis=alt.Axis(values=[start, 0, end])),
    y=alt.Y('tan(x)', axis=alt.Axis(values=[-math.sqrt(3), 0, math.sqrt(3)]))
)
# [END earthengine__apidocs__ee_array_tan]
